minimesos/src/main/java/com/containersol/minimesos/MesosCluster.java

```diff
@@ -62,6 +62,60 @@ public MesosCluster(ClusterArchitecture clusterArchitecture) {
 
         this.containers = clusterArchitecture.getClusterContainers().getContainers();
         clusterId = Integer.toUnsignedString(new SecureRandom().nextInt());
+        for (AbstractContainer container : containers) {
+            container.setClusterId(clusterId);
+        }
+    }
+
+    /**
+     * Recreate a MesosCluster object based on an existing cluster ID.
+     *
+     * @param clusterId the cluster ID of the cluster that is already running
+     */
+    public MesosCluster(String clusterId) {
+        this.clusterId = clusterId;
+
+        List<Container> containers = dockerClient.listContainersCmd().exec();
+
+        Collections.sort(containers, (c1,c2) -> Long.compare(c1.getCreated(), c2.getCreated()));
+
+        for (Container container : containers) {
+            for (String name : container.getNames()) {
+                if (name.matches("^/minimesos-.+-" + clusterId + "-\\d+$")) {
```

Comment:
To enforce naming convention, should we put composing and parsing containers' names to a separated class?